Add serialization for `LogicalCopyDatabase` operator

Flogex opened the pull request:
This PR adds serialization for the `LogicalCopyDatabase` operator and the `CopyDatabaseInfo` struct which is a member of `LogicalCopyDatabase`.

I first refactored `CopyDatabaseInfo` to enable this:
- I moved `CopyDatabaseInfo` to a separate file in the _parser/parsed_data_ directory. I hope this is the right place.
- `CopyDatabaseInfo` now inherits from `ParseInfo` to be in line with similar structs.
- Because a `Catalog` cannot be serialized, `CopyDatabaseInfo` now holds the target catalog name as a string. The actual `Catalog` instance gets resolved in `PhysicalCopyDatabase::GetData`. More on that below.
- I removed `CopyDatabaseInfo::from_database` because it is not used after the binding is done. All information required to execute the operator are in `target_database` and `entries`.

Serialization:
- I added entries for `CopyDatabaseInfo` and `LogicalCopyDatabase` in _parse\_info.json_ and _logical\_operator.json_, respectively, and regenerated the C++ code.
- I added a private constructor to `LogicalCopyDatabase` that takes a `ParseInfo` pointer which is then casted to `CopyDatabaseInfo`. This is needed because the auto-generated deserialization code does not do such a conversion automatically as far as I understand.

Is there a place where I should add tests for the serialization or is this all covered implicitly by the SQLLogic tests?

Resolving the catalog in `PhysicalCopyDatabase::GetData`:
I was not sure were the best place is to resolve the `CopyDatabaseInfo::target_database` string to an actual `Catalog` instance. For example, `LogicalCreateIndex` does the binding in the constructor. `LogicalCreateTable` stores a `BoundCreateTableInfo` and does the binding in a constructor that takes a`CreateInfo`.
I decided to go a similar route as, for example, `PhysicalCreatesSchema` and `PhysicalAlter` that resolve the catalog in `GetData`. Please let me know if you have another preferred way of doing this.

Flogex committed:
Refactor LogicalCopyDatabase and Bind for CopyDatabaseStatement

Mytherin commented:
Thanks for the PR! LGTM - serialization should be automatically tested by the SQLLogicTest as long as there are `COPY FROM DATABASE` queries being run with `PRAGMA enable_verification` enabled.

Mytherin commented on src/include/duckdb/parser/parsed_data/copy_database_info.hpp:
```diff
@@ -0,0 +1,40 @@
+	}
+
+	// The destination database to which catalog entries are being copied
+	std::string target_database;
```
nit: please avoid the `std::` prefix and use our own string type

Flogex replied:
I've removed the prefix.

Flogex committed:
Implement (de)serialization for LogicalCopyDatabase and CopyDatabaseInfo

Flogex committed:
Enable verification for some COPY DATABASE tests

Flogex commented:
I have added `PRAGMA enable_verification` to some of the `COPY DATABASE` tests to make sure the serialization is tested. (I didn't add it to the test that contains a call to `nextval` of a sequence because then another verifier would increment the sequence, leading to a wrong result in that test.)

I also rebased this PR to make the git history a bit cleaner. The old history is here: https://github.com/duckdb/duckdb/commits/5ad2641a38c24c7157014fb1a3a65dc441194194/

Mytherin commented:
Thanks!
